src/main/preload/externalAPI.ts

```diff
@@ -0,0 +1,459 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {contextBridge, ipcRenderer, webFrame} from 'electron';
+
+import {DesktopAPI} from 'types/api/main';
+
+import {
+    NOTIFY_MENTION,
+    IS_UNREAD,
+    UNREAD_RESULT,
+    SESSION_EXPIRED,
+    REACT_APP_INITIALIZED,
+    USER_ACTIVITY_UPDATE,
+    BROWSER_HISTORY_PUSH,
+    APP_LOGGED_IN,
+    APP_LOGGED_OUT,
+    GET_VIEW_INFO_FOR_TEST,
+    DESKTOP_SOURCES_RESULT,
+    VIEW_FINISHED_RESIZING,
+    CALLS_JOIN_CALL,
+    CALLS_JOINED_CALL,
+    CALLS_LEAVE_CALL,
+    DESKTOP_SOURCES_MODAL_REQUEST,
+    CALLS_WIDGET_SHARE_SCREEN,
+    CALLS_ERROR,
+    CALLS_JOIN_REQUEST,
+    GET_IS_DEV_MODE,
+    TOGGLE_SECURE_INPUT,
+    GET_APP_INFO,
+    REQUEST_BROWSER_HISTORY_STATUS,
+    BROWSER_HISTORY_STATUS_UPDATED,
+    NOTIFICATION_CLICKED,
+    CALLS_WIDGET_RESIZE,
+    CALLS_WIDGET_CHANNEL_LINK_CLICK,
+    CALLS_LINK_CLICK,
+    CALLS_POPOUT_FOCUS,
+    GET_DESKTOP_SOURCES,
+    MENTIONS_RESULT,
+    LEGACY_OFF,
+} from 'common/communication';
+
+contextBridge.exposeInMainWorld('desktopAPI', {
```

Comment:
So I actually wanted to use `desktop` instead of `desktopAPI` to expose all of the methods, but legacy versions of the Web App expect there to be different fields in that object.

However thinking about it, I might still be able to make that work if I tell the types to expect that field. Let me play with that, might be fixable :)

As for the names, I could add something else but if it's just the confusion between the two it's literally because of legacy support (there's going to be a lot of that :P)

Reply:
And now I remember why. Any object put onto the `window` by the `contextBridge` is read-only, so we can't write the version number to it. And older versions of the Web App will always try to do that.

We'll be ditching `window.desktop` permanently with the legacy code removal I'd say, so the confusion won't be there permanently.

Reply:
What's the difference between `desktopAPI.js` and `externalAPI.ts` then? That's more what's confusing me here

Reply:
Ahh. `desktopAPI.js` is for the internal stuff. That's an easy fix, I'll just rename it `internalAPI.js` :P